drivers/serial/uart_ra.c

```diff
@@ -345,16 +411,257 @@ static int uart_ra_init(const struct device *dev)
 		return ret;
 	}
 
+#ifdef CONFIG_UART_INTERRUPT_DRIVEN
+	ret = config->irq_config_func(dev);
+	if (ret != 0) {
+		return ret;
+	}
+#endif
+
 	return 0;
 }
 
+#ifdef CONFIG_UART_INTERRUPT_DRIVEN
+
+static bool uart_ra_irq_is_enabled(const struct device *dev,
+				     uint32_t irq)
+{
+	return !!(uart_ra_read_8(dev, SCR) & irq);
```

Comment:
no need for `!!` here, the compiler already does this for you since the return type is bool

Reply:
It makes sense. I fixed it.